src/rsp_restspawner/spawner.py

```diff
@@ -75,26 +75,29 @@ def from_sse(cls, sse: ServerSentEvent, progress: int) -> SpawnEvent:
         sse
             Event from the lab controller.
         progress
-            Current progress percentage. Parsing of the progress events that
-            communicate this must be done outside of this class.
+            Current progress percentage, if the event doesn't specify one.
         """
+        try:
+            data = sse.json()
+            if not (set(data.keys()) <= {"message", "progress"}):
+                raise ValueError("Invalid key in SSE data")
+            if "progress" in data:
+                progress = int(data["progress"])
+                if progress < 0 or progress > 100:
```

Comment:
Don't you want this outdented to also validate progress if it was passed in and not part of the message?